google-cloud-clients/google-cloud-contrib/google-cloud-logging-logback/src/main/java/com/google/cloud/logging/logback/MDCEventEnhancer.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.logging.logback;
+
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import com.google.cloud.logging.LogEntry;
+import java.util.Map;
+
+/**
+ * MDCEventEnhancer takes values found in the MDC property map and adds them as labels to the {@link
+ * LogEntry}. This {@link LoggingEventEnhancer} is turned on by default. If you wish to filter which
+ * MDC values get added as labels to your {@link LogEntry}, extend this class and override {@link
+ * MDCEventEnhancer#labelFilter(Map.Entry)} accordingly. The default behavior is to add all entries
+ * as labels.
+ */
+public class MDCEventEnhancer implements LoggingEventEnhancer {
```

Comment:
Do you think it's worthwhile to maintain this extra surface area moving forward? It seems like the `LoggingEventEnhancer` interface is simple enough for someone to implement a filtered label enhancer. The user now extends a base class and overrides one method vs. implementing one interface method (which looks like it's a functional interface to begin with).

Reply:
I think so. Only because if we don't expose this as a `LoggingEventEnhancer` I can't think of a good way to give users a hook into this without breaking behavior and removing automatically adding values as labels. If we are okay with changing that behavior, that is for sure another option.

Reply:
We could still have this class as a non-public implementation detail of that continues to implement the default behavior. By maintaining additional surface area, I mean encouraging people to extend this class vs. writing their own custom `LoggingEventEnhancer` implementation from scratch.

Reply:
Ah, gottcha. I like that, good idea.